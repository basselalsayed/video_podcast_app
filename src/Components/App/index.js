import React, { useEffect } from 'react';
import { connect } from 'react-redux';

import { getRss, isDirectional } from '../../Helpers';
import classes, { left, row } from './App.module.css';

import { Header, Feeds, Video } from '../';
import { Spinner } from 'react-bootstrap';

import { ARROW_KEYS } from '../../Constants';

const [UP, DOWN, LEFT, RIGHT] = ARROW_KEYS;

const AppBase = ({
  arrowPress,
  feed: { title, description, link },
  loading,
  setFeed,
  setLoading,
}) => {
  const rssHandler = async () => {
    setLoading(true);

    const response = await getRss();

    setFeed(response);
    setLoading(false);
  };

  const handleKeyDown = e => {
    if (isDirectional(e.key)) e.preventDefault();

    switch (e.key) {
      case UP:
        return arrowPress(UP);
      case DOWN:
        return arrowPress(DOWN);
      case LEFT:
        return arrowPress(LEFT);
      case RIGHT:
        return arrowPress(RIGHT);
      default:
        break;
    }
  };

  useEffect(() => {
    window.addEventListener('keydown', handleKeyDown);
    rssHandler();
  }, []);

  return (
    <div className={classes.App}>
      <Header
        title={title || 'Welcome'}
        description={description || 'Lorem Ipsum'}
        link={link}
        rssHandler={rssHandler}
      />
      {loading ? (
        <Spinner animation='border' />
      ) : (
        <div className={row}>
          <div className={left}>
            <Feeds />
          </div>

          <Video />
        </div>
      )}
    </div>
  );
};

const mapStateToProps = state => ({
  loading: state.loading,
  feed: state.feed,
  nowPlaying: state.nowPlaying,
});

const mapDispatchToProps = dispatch => ({
  arrowPress: type => dispatch({ type }),
  setLoading: loading =>
    dispatch({ type: 'SET_LOADING', payload: { loading } }),
  setFeed: feed => dispatch({ type: 'SET_FEED', payload: { feed } }),
});

const App = connect(mapStateToProps, mapDispatchToProps)(AppBase);

export { App };
