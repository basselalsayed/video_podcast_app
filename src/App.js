import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import classes, { column, row } from './App.module.css';

import { newRss } from './api';
import { Header, Feeds, Video } from './Components';
import { Spinner } from 'react-bootstrap';
import { VideoContext } from './Components/Video/context';
import withClass from './hoc/withClass';

const App = ({
  loadingRedux,
  feedRedux,
  nowPlayingRedux,
  setLoadingRedux,
  setFeedRedux,
}) => {
  // const [loading, setLoading] = useState(false);
  // const [feed, setFeed] = useState({});
  const [nowPlaying, setNowPlaying] = useState(
    'http://pmd.cdn.turner.com/cnn/big/cnn10/2020/06/13/ten-0615.cnn_3275653_ios_1240.mp4',
  );

  // const rssHandler = async () => {
  //   setLoading(true);

  //   const response = await newRss();

  //   setFeed(response);
  //   setLoading(false);
  // };

  const rssHandlerRedux = async () => {
    setLoadingRedux(true);

    const response = await newRss();

    setFeedRedux(response);
    setLoadingRedux(false);
  };

  useEffect(() => {
    rssHandlerRedux();
  }, []);

  return (
    <VideoContext.Provider value={{ nowPlaying, setNowPlaying }}>
      <Header
        title={feedRedux.title || 'Welcome'}
        description={feedRedux.description || 'Lorem Ipsum'}
        rssHandler={rssHandlerRedux}
      />
      {loadingRedux ? (
        <Spinner animation='border' />
      ) : (
        <div className={row}>
          <div className={column}>
            <Feeds feeds={feedRedux.items} />
          </div>
          <div className={column}>
            <Video />
          </div>
        </div>
      )}
    </VideoContext.Provider>
  );
};

const mapStateToProps = state => ({
  loadingRedux: state.loading,
  feedRedux: state.feed,
  nowPlayingRedux: state.nowPlaying,
});

const mapDispatchToProps = dispatch => ({
  setLoadingRedux: loading =>
    dispatch({ type: 'SET_LOADING', payload: { loading } }),
  setFeedRedux: feed => dispatch({ type: 'SET_FEED', payload: { feed } }),
});

const StyledApp = withClass(App, classes.App);

export default connect(mapStateToProps, mapDispatchToProps)(StyledApp);
