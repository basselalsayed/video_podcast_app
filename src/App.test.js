import App from './App';

describe('<App />', () => {
  const wrapper = shallow(<App />);

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
