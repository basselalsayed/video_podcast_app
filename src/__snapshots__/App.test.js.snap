// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<div
  className="App"
>
  <Header
    description="Lorem Ipsum"
    title="Welcome"
  />
  <p>
    false
  </p>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    Update Feed
  </Button>
</div>
`;
